### PR TITLE
Feature/effects, third person movement

### PR DESCRIPTION
**Pull request by Dotsially**
This is 3 pull requests into one.
- Third person movement.
- Working shotgun recoil.
- Shotgun class. 
- Shotgun muzzle and reload effects.
and more.

1. Merging

- [x] Did you select the dev branch?

   

- [X]  Are all scripts placed within Assets/Scripts and organized into logical categories using PascalCase (e.g., Core, Player, Level)?

2. File Structure

- [x] Are all scripts placed within Assets/Scripts and organized into logical categories using PascalCase (e.g., Core, Player, Level)?
- [x] Are assets like materials, prefabs, and textures named using lower_snake_case (e.g., diamond_pickaxe.png)?

- [x] Did the changes avoid unnecessary scene modifications? If new changes were needed for testing, were they done in Scenes/Dev/ and then discarded?

3. Code

- [X] Do private fields start with an underscore and use camelCase (e.g., _privateVariable)?
- [X] Are static readonly variables named using UPPER_SNAKE_CASE?
- [X] Are enum types also in UPPER_SNAKE_CASE?
- [X] Are function and variable names clear, concise, and descriptive without using unnecessary abbreviations?
- [ ] Are all public functions and fields documented with clear summaries where needed
- [ ] Does the code maintain a strict hierarchy, ensuring objects do not fetch references themselves or their parents in the hierarchy?
- [ ] Are references passed down through dependency injection as needed?Does each component only depend on its parent in the hierarchy?
- [ ] Are there any unnecessary Debug.Logs?

### CODE REVIEW

**Review by IcyKoala — CHANGES_REQUESTED** [review]
Few small things, but looks great in general
- Not sure if it's a problem or not, but the PlayerController under Player seems to very slowly rotate when not moving.
- Can you check the material names because they aren't using lower_snake_case atm.
And my other comments

File: Project/Assets/Scripts/Player/MoveState.cs [hunk 1]
```diff
@@ -48,6 +53,12 @@ public void OnStateFixedProcessing()
         // Preserve the y component
         float y = _playerRigidbody.velocity.y;
 
+        //Move player model and its spine according to the camera
+        _playerModel.transform.rotation = Quaternion.Euler(0,  Camera.main.transform.rotation.eulerAngles.y, 0);
+        Quaternion lookRotation = Quaternion.LookRotation((Camera.main.transform.position + Camera.main.transform.forward * 10.0f) - _playerSpine.position);
+        _playerSpine.rotation = math.slerp(_playerSpine.rotation, lookRotation, 0.9f);
```
[the remark this answers is not in the document]
Comment (Dotsially) [reply Project/Assets/Scripts/Player/MoveState.cs:59]: This isn't the rate of turning it just so the player turns with the camera. Will not be changed.

File: Project/Assets/Scripts/SoundEffectManager.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+public class SoundEffectManager : MonoBehaviour
+{
+    public static SoundEffectManager Instance;
+    [Range(0.1f, 0.5f)]
+    public float pitchMultiplier = 0.1f;
+    [Range(0.01f, 0.3f)]
+    public float volumeMultiplier = 0.01f;
+    
+    public AudioSource soundEffectObject;
```
[the remark this answers is not in the document]
Comment (Dotsially) [reply Project/Assets/Scripts/SoundEffectManager.cs]: Done

File: Project/Assets/Scripts/FollowPosition.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+public class FollowPosition : MonoBehaviour
+{
+    public Transform playerTransform;
+    public Vector3 offset;
```
[the remark this answers is not in the document]
Comment (Dotsially) [reply Project/Assets/Scripts/FollowPosition.cs]: Done

**Review by IcyKoala — APPROVED** [review]

**Pull request merged**